qiskit_nature/algorithms/initial_points/mp2_initial_point.py

```diff
@@ -0,0 +1,363 @@
+        else:
+            if not isinstance(self._excitation_list, list):
+                raise QiskitNatureError(
+                    "Cannot compute. `excitation_list` has not been set directly or via `ansatz`."
+                )
+
+        self._corrections = self._compute_corrections()
+
+    def _compute_corrections(
+        self,
+    ) -> dict[str, tuple[float, float]]:
+        """Compute the MP2 coefficients and energy corrections.
+
+        Tuples of the coefficients and energy corrections are stored in a dictionary with a string
+        of the corresponding excitation. This dictionary isn't directly exposed to the user, but is
+        retained for internal clarity and validation. Non-double excitations will have zero
+        coefficient and energy correction.
+
+        Returns:
+            Dictionary with MP2 coefficients and energy_corrections for each excitation.
+        """
+        corrections = {}
+        for excitation in self._excitation_list:
+            if len(excitation[0]) == 2:
+                # Compute MP2 corrections using double excitations.
+                coefficient, energy_correction = self._compute_correction(excitation)
+            else:
+                # No corrections for single, triple, and higher excitations.
+                coefficient, energy_correction = 0.0, 0.0
+
+            corrections[str(excitation)] = (coefficient, energy_correction)
+
+        return corrections
+
+    def _compute_correction(
+        self, excitation: tuple[tuple[int, ...], tuple[int, ...]]
+    ) -> tuple[float, float]:
+        """Compute the MP2 coefficient and energy correction given a double excitation.
+
+        Each double excitation indexed by ::math`i,a,j,b` has a correction coefficient,
+
+        ..math::
+            c_{i,a,j,b} = -\\frac{2 T_{i,a,j,b} - T_{i,b,j,a}}{E_b + E_a - E_i - E_j},
+
+        where ::math::`E` is the orbital energy and ::math::`T` is the integral matrix.
+        And an energy correction,
+
+        ..math..
+            correction = E_{i, a, j, b} = c_{i,a,j,b} T_{i,a,j,b}.
+
+        These computations use molecular orbitals, but the indices used in the excitation
+        information passed in and out use the block spin orbital numbering common to Qiskit Nature:
+          - ::math::`\\alpha` runs from ``0`` to ``num_molecular_orbitals - 1``,
+          - ::math::`'\\'beta` runs from ``num_molecular_orbitals`` to
+            ``num_molecular_orbitals * 2 - 1``.
+
+        Args:
+            excitations: List of excitations.
+
+        Returns:
+            Coefficient and energy correction for a given double excitation.
+        """
+        integral_matrix = self._integral_matrix
+        orbital_energies = self._orbital_energies
+        threshold = self._threshold
+
+        # Infer the number of molecular orbitals from the MO matrix.
+        num_molecular_orbitals: int = integral_matrix.shape[0]
+
+        i = excitation[0][0] % num_molecular_orbitals
```

Comment:
A suggestion for the future when extending this to unrestricted spin cases:
```suggestion
        a_or_b_i, i = divmod(excitation[0][0], num_molecular_orbitals)
```
`a_or_b_i` will then be either 0 or 1 indicating which set of 2-body terms to use.

Reply:
Thanks! Added as a suggestion to #645.